drivers/dma/dma_silabs_siwx917.c

```diff
@@ -239,12 +239,12 @@ static int dma_channel_config(const struct device *dev, RSI_UDMA_HANDLE_T udma_h
 		channel_control.rPower = ARBSIZE_1024;
 	}
 	/* Obtain the number of transfers */
-	config->head_block->block_size /= config->source_data_size;
-	if (config->head_block->block_size >= DMA_MAX_TRANSFER_COUNT) {
+	if ((config->head_block->block_size / config->source_data_size) >= DMA_MAX_TRANSFER_COUNT) {
```

Comment:
You're evaluating `config->head_block->block_size / config->source_data_size` three times in this function. Seems like it'd be better to do the calculation just once and store the result in a local helper variable.